falcon/api.py

```diff
@@ -103,6 +103,10 @@ def process_response(self, req, resp, resource, req_succeeded)
             to use in lieu of the default engine.
             See also: :ref:`Routing <routing>`.
 
+        router_search_method (str, optional): The name of a custom router method
```

Comment:
How does this compare to simply defining a new `search()` method for the router interface and deprecating `find()`? Does the method name need to be configurable?